src/sync/core-sync-state.js

```diff
@@ -134,6 +147,10 @@ export class CoreSyncState {
   insertPreHaves(peerId, start, bitfield) {
     const peerState = this.#getPeerState(peerId)
     peerState.insertPreHaves(start, bitfield)
+    this.#preHavesLength = Math.max(
+      this.#preHavesLength,
+      peerState.preHavesBitfield.lastSet(start + bitfield.length * 32) + 1
```

Comment:
It looks like `32` was changed to `128` upstream. Just double checking: is this `32` okay here? (Broadly, I'm a little surprised that we need to do this multiplication at all, but I think it's fine.)

Reply:
It's slightly confusing, in that here `bitfield` refers to a Uint32Array, rather than an instance of RemoteBitfield. For the core length, we need the length in bits (because one bit = one block in the bitfield), and since each item in the array is 32 bits (because it's a Uint32Array), then we need the `length * 32`.